### PR TITLE
Table sort feature

### PR DESCRIPTION
**Pull request by MartinezBaker**

### CODE REVIEW

File: crypto-app/src/pages/Coins/Coins.js [hunk 1]
```diff
@@ -1,38 +1,45 @@
 import React from 'react';
 import InfiniteScroll from "react-infinite-scroll-component";
 import axios from "axios";
+import { setSortIcon } from '../../utils/FontAwsomeutil'
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: import { setSortIcon } from 'utils/FontAwsomeutil'

File: crypto-app/src/pages/Coins/Coins.js [hunk 2]
```diff
@@ -1,117 +1,238 @@
...
-      const addCoins = [...this.state.coins, ...data]
-      setTimeout(() => {
-        this.setState({
-          coins: addCoins,
-          isLoading: false
-        })
-      }, 500)
-    }catch(error){
-      this.setState({isLoading: false, errMess:"Could not load more coins!"})
+      if(data === !![].length) {
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: if(!data.length)

File: crypto-app/src/pages/Coins/Coins.js [hunk 3]
```diff
@@ -1,117 +1,238 @@
...
-      this.getCoins()
+    this.getCoins();
   }
-  render(){
+  handleSort = (sortType) => {
+    const newSort = Object.entries(this.state.sort).map((entry) => {
+      const [key, value] = entry
+      if(key === sortType) {
+        return {
+          [key]: value !== true ? true : false
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: [key]: value !== true

File: crypto-app/src/pages/Coins/Coins.js [hunk 4]
```diff
@@ -1,117 +1,238 @@
...
+       coinList = coinList.sort((a, b) =>
+        a.name.toUpperCase() < b.name.toUpperCase() ? -1 : 1
+      );
+    }
+    if (sortName === false) {
+       coinList = coinList.sort((a, b) =>
+        b.name.toUpperCase() < a.name.toUpperCase() ? -1 : 1
+      );
+    }
+
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: if(sortValue === true){
 coinList = coinList.sort((a, b) =>
        b[sortValue].toUpperCase() < a[sortValue].toUpperCase() ? -1 : 1
      );
}

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: crypto-app/src/pages/Coins/Coins.js (hunk 3 above)
Comment (MartinezBaker) [reply crypto-app/src/pages/Coins/Coins.js]: I'm not really clear on what you are asking me to do different here..

**Pull request merged**